File: src/test/java/org/dxworks/githubminer/service/repository/commits/GithubCommitServiceIT.java
package org.dxworks.githubminer.service.repository.commits;

import org.dxworks.githubminer.dto.response.repository.commits.RepoCommit;
import org.dxworks.githubminer.utils.TestUtils;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class GithubCommitServiceIT {

    private GithubCommitService githubCommitService = new GithubCommitService("MarioRivis", "SimpleRegistrationExample", TestUtils.getGithubCredentials());

    @Test
    void testGetAllCommits() {
        List<RepoCommit> allCommits = githubCommitService.getAllCommits();
        assertTrue(allCommits.stream().allMatch(repoCommit -> repoCommit instanceof RepoCommit));
        assertEquals(7, allCommits.size());
    }
}

File: github-miner-core/src/test/kotlin/org/dxworks/githubminer/service/repository/commits/GithubCommitServiceIT.kt
package org.dxworks.githubminer.service.repository.commits

import org.dxworks.githubminer.dto.response.repository.commits.RepoCommit
import org.dxworks.githubminer.TestUtils.githubCredentials
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class GithubCommitServiceIT {
    private val githubCommitService = GithubCommitService("apache", "kafka", githubTokens = githubCredentials)

    @Test
    fun testGetAllCommits() {
        val allCommits = githubCommitService.allCommits
        Assertions.assertTrue(allCommits.stream().allMatch { repoCommit: RepoCommit? -> repoCommit is RepoCommit })
        Assertions.assertEquals(7, allCommits.size)
    }
}
